openshift_tools/monitoring/zagg_sender.py

```diff
@@ -27,22 +27,60 @@
     zs.send_metrics()
 """
 
-from openshift_tools.monitoring.zagg_client import ZaggClient
 from openshift_tools.monitoring import pminfo
 from openshift_tools.monitoring.metricmanager import UniqueMetric
+from openshift_tools.monitoring.zagg_client import ZaggClient
+from openshift_tools.monitoring.zagg_common import ZaggConnection
+import yaml
 
 class ZaggSender(object):
     """
     collect and create UniqueMetrics and send them to Zagg
     """
 
-    def __init__(self, host, zagg_connection):
+    def __init__(self, host=None, zagg_connection=None):
         """
         set up the zagg client, pcp_metrics and unique_metrics
         """
-        self.zaggclient = ZaggClient(zagg_connection=zagg_connection)
-        self.host = host
         self.unique_metrics = []
+        self.config = None
+        self.config_file = '/etc/openshift_tools/zagg_client.yaml'
+
+        if not host:
+            host = self.get_default_host()
+
+        if not zagg_connection:
+            zagg_connection = self.get_default_zagg_connecton()
+
+        self.host = host
+        self.zaggclient = ZaggClient(zagg_connection=zagg_connection)
+
+    def parse_config(self):
+        """ parse default config file """
+
+        if not self.config:
```

Comment:
This should probably check whether self.config_file exists as well.  If it does not exist it will throw an exception.